### PR TITLE
feat(client): change of container & anchor's css on project hover

### PR DESCRIPTION
**Pull request by lalshubham**
On User's project hover, we can change the <a> View's background color, border and text color. 
Also, on project hover, we can show some rounded border on "portfolio-container".

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

Here's the video preview of what I'm trying to demonstrate:

https://github.com/freeCodeCamp/freeCodeCamp/assets/111045472/c36fbf12-5281-4a79-ad8a-63479f9c27f1

### CODE REVIEW

**Review by Sboonny — COMMENTED** [review]
I am in conflict about the animation 🤔. But I will let a designer discuss it instead.

If we are updating that page, can you:

- Add these values to `.portfolio-container` in `@media (min-width: 995px)`

```css
.portfolio-container {
    grid-template-columns: 450px 1.2fr;
    grid-template-rows: minmax(65px, 0.1fr) minmax(125px, 1fr) 75px;
}
```

- Can you replace the `<FullWidthRow>` component with Fragment in the file below

https://github.com/freeCodeCamp/freeCodeCamp/blob/5b3ecb703559d8c1c08647b0d8b93278f490ebf6/client/src/components/profile/components/portfolio-projects.tsx#LL22C5-L22C19

**Review by Sboonny — COMMENTED** [review]
I am in conflict about the animation 🤔. But I will let a designer discuss it instead.

If we are updating that page, can you:

- Add these values to `.portfolio-container` in `@media (min-width: 995px)`

```css
.portfolio-container {
    grid-template-columns: 450px 1.2fr;
    grid-template-rows: minmax(65px, 0.1fr) minmax(125px, 1fr) 75px;
}
```

- Can you replace the `<FullWidthRow>` component with Fragment in the file below

https://github.com/freeCodeCamp/freeCodeCamp/blob/5b3ecb703559d8c1c08647b0d8b93278f490ebf6/client/src/components/profile/components/portfolio-projects.tsx#LL22C5-L22C19

**Review by Sboonny — COMMENTED** [review]
I am in conflict about the animation 🤔. But I will let a designer discuss it instead.

If we are updating that page, can you:

- Add these values to `.portfolio-container` in `@media (min-width: 995px)`

```css
.portfolio-container {
    grid-template-columns: 450px 1.2fr;
    grid-template-rows: minmax(65px, 0.1fr) minmax(125px, 1fr) 75px;
}
```

- Can you replace the `<FullWidthRow>` component with Fragment in the file below

https://github.com/freeCodeCamp/freeCodeCamp/blob/5b3ecb703559d8c1c08647b0d8b93278f490ebf6/client/src/components/profile/components/portfolio-projects.tsx#LL22C5-L22C19

**Review by Sboonny — COMMENTED** [review]
I am in conflict about the animation 🤔. But I will let a designer discuss it instead.

If we are updating that page, can you:

- Add these values to `.portfolio-container` in `@media (min-width: 995px)`

```css
.portfolio-container {
    grid-template-columns: 450px 1.2fr;
    grid-template-rows: minmax(65px, 0.1fr) minmax(125px, 1fr) 75px;
}
```

- Can you replace the `<FullWidthRow>` component with Fragment in the file below

https://github.com/freeCodeCamp/freeCodeCamp/blob/5b3ecb703559d8c1c08647b0d8b93278f490ebf6/client/src/components/profile/components/portfolio-projects.tsx#LL22C5-L22C19

**Comment by CERTIFIED2003** [comment]
Yes sure @Sboonny. 
Doing it soon!

**Review by Sboonny — APPROVED** [review]
#### :sparkles: :tada: **AWESOME!** :tada: :sparkles:

 Hi @CERTIFIED2003, thanks for this pull request and for contributing to the code-base for the first time. We are looking forward to more contributions from you in the future.

 Cheers & happy contributing!

**Comment by CERTIFIED2003** [comment]
Hey @Sboonny, thanks for that. It was my first open-source PR. That's means a lot to me and yeah I am preparing for future contributions too. Thanks again🙏

**Comment by CERTIFIED2003** [comment]
Final Preview:

https://github.com/freeCodeCamp/freeCodeCamp/assets/111045472/1efeb7a0-34cb-40a4-a0f5-a0dfb17baba2

**Comment by bbsmooth** [comment]
I'm not a designer, so take this with a grain of salt, but I'm not thrilled with rounding the border on hover. Perhaps bolding it? Giving it a different color? Both of these? I think either/both of these options make it more obvious that the entire container is clickable.

But I am going to more strongly argue that the background color on the button should not change unless the button itself is hovered. Buttons are a distinct element and their visible state shouldn't change unless they are acted upon.

**Comment by CERTIFIED2003** [comment]
Appreciate your words @bbsmooth.
I understand your explanation about the `border-radius` concept, but I would like to express an alternative viewpoint regarding your second opinion. Consider this: Given that the entire container serves as the link, it does make sense that hovering over the `container` would trigger changes to the button's styling.

Furthermore, I am forward to hear more from you, sir.

**Comment by ahmadabdolsaheb** [comment]
Thank you for your interest in improving the design of the project cards.
We do not have a holistic approach to animations yet, but a tasteful short one would not hurt.
For stretched link cards patterns such as the portfolio projects, only the color of the button should be changed. 

Actionable items:
1: make the button (link button) look like the other buttons on the page.
2: reverse its colors on hover.

Here is the general solution for your reference: [stretched-link/](https://getbootstrap.com/docs/5.0/helpers/stretched-link/)

File: client/src/components/profile/components/portfolio-projects.tsx [hunk 1]
```diff
@@ -19,7 +18,7 @@ export const PortfolioProjects = ({
     return null;
   }
   return (
-    <FullWidthRow>
+    <>
```
Comment (ahmadabdolsaheb) [inline client/src/components/profile/components/portfolio-projects.tsx:21]: This change is out of scope for this pr. When reverting this change, please alter the css accordingly to make sure the button does not go out of the card.

**Comment by ahmadabdolsaheb** [comment]
Thanks for implementing the changes to improve the look of the project cards. If you wish to improve the card further, I encourage you to open an issue so we could finalize the spec before proceeding with the implementations.

File: client/src/components/profile/components/portfolio-projects.tsx (hunk 1 above)
Comment (Sboonny) [reply client/src/components/profile/components/portfolio-projects.tsx:21]: This change is my suggestion, if it is reverted it would break the layout in medium layout

Comment (ahmadabdolsaheb) [reply client/src/components/profile/components/portfolio-projects.tsx:21]: In that case, LGMT. The layout of the card can improved further as there are a lot of unused white space. If you are interested we could tackle it in a separate issue/pr.

**Review by ahmaxed — APPROVED** [review]
